cmd/staking.go

```diff
@@ -0,0 +1,156 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/spf13/cobra"
+)
+
+func stakingDelegateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delegate [validator-addr] [amount] [from]",
+		Args:  cobra.ExactArgs(3),
+		Short: "Delegate liquid tokens to a validator",
+		Long: strings.TrimSpace(
+			`Delegate an amount of liquid coins to a validator from your wallet.
+Example:
+$ lens tx staking delegate cosmosvaloper1sjllsnramtg3ewxqwwrwjxfgc4n4ef9u2lcnj0 1000stake mykey`,
+		),
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			var (
+				delAddr sdk.AccAddress
+				err     error
+			)
+
+			cl := config.GetDefaultClient()
+
+			amount, err := sdk.ParseCoinNormalized(args[1])
+			if err != nil {
+				return err
+			}
+
+			if cl.KeyExists(args[2]) {
+				delAddr, err = cl.GetKeyAddress()
```

Comment:
So this will pull the default key, not the one given by args[2]. A couple of options: set the default key on the client (i.e. `cl.Config.Key == args[2]`) before trying to get `GetKeyAddress` or use/create a function like `GetKeyAddress(name string)` and change current one to `GetDefaultAddress()`